src/docs/sdk/performance/dynamic-sampling-context.mdx

```diff
@@ -0,0 +1,168 @@
+As a mental model:
+The head transaction in a trace determines the Dynamic Sampling Context for all following transactions in that trace.
+No information can be changed, added or deleted after the first propagation.
+
+### SDKs
+
+SDKs are responsible for propagating **Dynamic Sampling Context** across all applications that are part of a trace.
+This involves:
+
+1. Collecting the information that makes up the DSC **xor** extracting the DSC from incoming requests.
+2. Propagating DSC to downstream SDKs.
+3. Sending the DSC to Sentry via the `trace` envelope header.
+
+Because there are quite a few things to keep in mind for DSC propagation and to avoid every SDK running into the same problems, we defined a [unified propagation mechanism](#unified-propagation-mechanism) (step-by-step instructions) that all SDK implementations should be able to follow.
+
+## Baggage
+
+We chose `baggage` as the propagation mechanism for DSC. ([w3c baggage spec](https://www.w3.org/TR/baggage/))
+Baggage is a standard HTTP header with URI encoded key-value pairs.
+
+For the propagation of DSC, SDKs first read the DSC from the baggage header of incoming requests/messages.
+To propagate DSC to downstream SDKs/services, we create a baggage header (or modify an existing one) through HTTP request instrumentation.
+
+<Alert level="info">
+
+Other vendors might also be using the `baggage` header.
+If a `baggage` header already exists on an outgoing request, SDKs should aim to be good citizens by only **appending** Sentry values to the header.
+In the case that another vendor added Sentry values to an outgoing request, SDKs may overwrite those values.
+
+SDKs must not add other vendors' baggage from incoming requests to outgoing requests.
+Sentry SDKs only concern themselves with Sentry baggage.
+
+</Alert>
+
+The following is an example of what a baggage header containing Dynamic Sampling Context may look like:
+
+```
+baggage: other-vendor-value-1=foo;bar;baz, sentry-traceid=771a43a4192642f0b136d5159a501700, sentry-publickey=49d0f7386ad645858ae85020e393bef3; sentry-userid=Am%C3%A9lie, other-vendor-value-2=foo;bar;
+```
+
+See the [Payloads section](#payloads) for a complete list of key-value pairs that SDKs should propagate.
+
+## Payloads
+
+Dynamic Sampling Context is propagated via a baggage header and sent to Sentry via the `trace` envelope header.
+
+### Baggage-Header
+
+SDKs may set the following key-value pairs on baggage headers.
+While none of these values are strictly required, SDKs should make their best effort to add as many of them to the baggage header as possible when starting a trace.
+
+Some values are marked as "required for Dynamic sampling to work".
+Nothing bad is going to happen when these values are missing, however, Dynamic Sampling will not work without them.
+
+- `sentry-traceid` - **Required for Dynamic Sampling to work** - The original trace ID as generated by the SDK.
```

Comment:
All of these are required for dynamic sampling to work. Trace sampling rules match on release/environment from this baggage header, not the event content

Reply:
Are they *really* required though? What if a user doesn't have an environment set but a user id? Do we want to prevent users from using DS when they don't have environments?

Reply:
i mean that if the sdk knows an environment, it needs to be in that baggage. in that sense it is required. if not then dyn sampling is broken, because the user will configure an environment and won't be able to sample by it

Reply:
Why is `traceid` needed? Is that not already in the `sentry-trace` header? That seems redundant and error prone.

Reply:
We put this in from the original dynamic sampling implementation (one with tracestate as the propogation mechanism) - it was just inherited from the old spec. 

Copy pasting an exchange from slack for some added context here:

Armin:
> but the need for sentry-traceid is not obvious to me

Abhi:
> It's for the dynamic sampling logic right? So we can peek in the envelope header to grab the trace_id without having to reach into the event payload. Otherwise how would relay know about the trace?

Armin: 
> relay would still need it in the `trace` header but sdk to sdk communication does not need to pass it in the baggage header. 
> basically we would treat it as special item in the baggage that is sent to relay but not between sdks as it’s already in `sentry-trace`
> **i worry about this desyncing**

I'd rather just keep `traceid` in the Dynamic Sampling Context (DSC) that propogated with baggage to simplify the implementation. Since baggage should be travelling alongside `sentry-trace` at all points, I can't see a a way that it would ever get out of sync. Having baggage and the envelope header have identical Dynamic Sampling Context makes it much easier to understand and debug + prevents the need for us to go back and change the SDKs yet again. I think the friction that comes from trying to drop `traceid` in just baggage specifically is not worth it - and it's not that hard for us to revisit if needed.

Reply:
> i mean that if the sdk knows an environment, it needs to be in that baggage. in that sense it is required. if not then dyn sampling is broken, because the user will configure an environment and won't be able to sample by it

@untitaker Yeah I see the problem now. Updated the wording to clarify: 67b98b2f75474dd52572263f246a84af65600039